src/Common/tests/gtest_resolve_pool.cpp

```diff
@@ -310,65 +338,77 @@ TEST_F(ResolvePoolTest, DuplicatesInAddresses)
     ASSERT_EQ(3, DB::CurrentThread::getProfileEvents()[metrics.discovered]);
 }
 
-void check_no_failed_address(size_t iteration, auto & resolver, auto & addresses, auto & failed_addr, auto & metrics)
+void check_no_failed_address(size_t iteration, auto & resolver, auto & addresses, auto & failed_addr, auto & metrics, auto deadline)
 {
     ASSERT_EQ(iteration, DB::CurrentThread::getProfileEvents()[metrics.failed]);
     for (size_t i = 0; i < 100; ++i)
     {
         auto next_addr = resolver->resolve();
+
+        if (now() > deadline)
+            break;
```

Comment:
```suggestion
        if (now() > deadline)
        {
            ASSERT_NE(i, 0);
            break;
        }
```

Just to make sure the check is done at least once.